instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/genai/GenAiClientMetrics.java

```diff
@@ -0,0 +1,123 @@
+import io.opentelemetry.instrumentation.api.instrumenter.OperationMetrics;
+import io.opentelemetry.instrumentation.api.internal.OperationMetricsUtil;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+/**
+ * {@link OperationListener} which keeps track of <a
+ * href="https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-metrics/#generative-ai-client-metrics">Generative
+ * AI Client Metrics</a>.
+ */
+public final class GenAiClientMetrics implements OperationListener {
+
+  private static final double NANOS_PER_S = TimeUnit.SECONDS.toNanos(1);
+
+  private static final ContextKey<State> GEN_AI_CLIENT_METRICS_STATE =
+      ContextKey.named("gen-ai-client-metrics-state");
+
+  private static final Logger logger = Logger.getLogger(DbClientMetrics.class.getName());
+
+  static final AttributeKey<String> GEN_AI_TOKEN_TYPE = stringKey("gen_ai.token.type");
+
+  /**
+   * Returns an {@link OperationMetrics} instance which can be used to enable recording of {@link
+   * GenAiClientMetrics}.
+   *
+   * @see
+   *     io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder#addOperationMetrics(OperationMetrics)
+   */
+  public static OperationMetrics get() {
+    return OperationMetricsUtil.create("gen_ai client", GenAiClientMetrics::new);
+  }
+
+  private final LongHistogram tokenUsage;
+  private final DoubleHistogram operationDuration;
+
+  private GenAiClientMetrics(Meter meter) {
+    LongHistogramBuilder tokenUsageBuilder =
+        meter
+            .histogramBuilder("gen_ai.client.token.usage")
+            .ofLongs()
+            .setUnit("{token}")
+            .setDescription("Measures number of input and output tokens used")
+            .setExplicitBucketBoundariesAdvice(GenAiMetricsAdvice.CLIENT_TOKEN_USAGE_BUCKETS);
+    GenAiMetricsAdvice.applyClientTokenUsageAdvice(tokenUsageBuilder);
+    this.tokenUsage = tokenUsageBuilder.build();
+    DoubleHistogramBuilder operationDurationBuilder =
+        meter
+            .histogramBuilder("gen_ai.client.operation.duration")
+            .setUnit("s")
+            .setDescription("GenAI operation duration")
+            .setExplicitBucketBoundariesAdvice(
+                GenAiMetricsAdvice.CLIENT_OPERATION_DURATION_BUCKETS);
+    GenAiMetricsAdvice.applyClientOperationDurationAdvice(operationDurationBuilder);
+    this.operationDuration = operationDurationBuilder.build();
+  }
+
+  @Override
+  public Context onStart(Context context, Attributes startAttributes, long startNanos) {
+    return context.with(
+        GEN_AI_CLIENT_METRICS_STATE,
+        new AutoValue_GenAiClientMetrics_State(startAttributes, startNanos));
+  }
+
+  @Override
+  public void onEnd(Context context, Attributes endAttributes, long endNanos) {
+    State state = context.get(GEN_AI_CLIENT_METRICS_STATE);
+    if (state == null) {
+      logger.log(
+          FINE,
+          "No state present when ending context {0}. Cannot record database operation metrics.",
```

Comment:
replace `database operation` with whatever is appropriate

Reply:
Oops, thanks fixed